apps/passport-client/src/loadInitialState.ts

```diff
@@ -43,20 +44,25 @@ export async function loadInitialState(): Promise<AppState> {
     self &&
     (!self.semaphore_v4_commitment || !self.semaphore_v4_pubkey)
   ) {
-    const semaphoreV4IdentityPCD = findIdentityV4PCD(pcds);
-    if (semaphoreV4IdentityPCD) {
-      await requestUpgradeUserWithV4Commitment(
-        appConfig.zupassServer,
-        await makeUpgradeUserWithV4CommitmentRequest(pcds)
-      );
-    }
+    const identityPCD = v3IdentityToPCD(identityV3);
+
+    await requestUpgradeUserWithV4Commitment(
+      appConfig.zupassServer,
+      await makeUpgradeUserWithV4CommitmentRequest(
+        new PCDCollection(await getPackages(), [identityPCD])
```

Comment:
Why is this making a fake temporary PCD collection?  There's one available in the `pcds` variable.  If there's a reason not to depend on PCDs, then maybe makeUpgradeUserWithV4CommitmentRequest should take a bare Identity as input?

Separate but related to this, I think it would be better to do this upgrade request farther down after state validation, so that we don't try to change things on the server until we know this client is in a valid state.

Reply:
> Why is this making a fake temporary PCD collection? There's one available in the pcds variable. If there's a reason not to depend on PCDs, then maybe makeUpgradeUserWithV4CommitmentRequest should take a bare Identity as input?

It is possible for a pcd collection to be empty, but for there to be an identity pcd in the client. It's not super well-defined, but I'd like to leave that for another time.

Reply:
It's reasonable to consider that it might be missing.  Might be less of an issue if this happened after state validation as I mentioned in my second point, which I think has a higher impact.  I'd still say that if what `makeUpgradeUserWithV4CommitmentRequest` needs is an Identity, not a PCD then there should simply be a version which takes that as an argument, rather than requiring wrapping it in a whole PCD collection, but that's about cleanliness not correctness.